sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -540,10 +564,25 @@ object DottyPlugin extends AutoPlugin {
       dottyVersion,
       fullLoader,
       libraryLoader,
-      Array(dottyLibrary, scalaLibrary),
+      libraryJars,
       compiler,
       all.toArray,
       None)
+  }
+}
+
+private class TopClassLoader(sbtLoader: ClassLoader) extends ClassLoader(null) {
```

Comment:
A documentation comment would be nice (maybe move/copy some part of the comment where TopClassLoader is used. It's also worth explaining why we pass `null` to the super-call.